src/pages/Main.tsx

```diff
@@ -1,7 +1,15 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import { Link, RouteComponentProps } from 'react-router-dom';
+import * as THREE from 'three';
 import { HistoryParams } from '../types/react-router';
 
+import CHUN from '../assets/tiles/Regular/Chun.png';
+import MAN1 from '../assets/tiles/Regular/Man1.png';
+import HATSU from '../assets/tiles/Regular/Hatsu.png';
+import NAN from '../assets/tiles/Regular/Nan.png';
+import PEI from '../assets/tiles/Regular/Pei.png';
+import SOU1 from '../assets/tiles/Regular/Sou1.png';
+
```

Comment:
This and all other Three related code. Move outside of page into a component in components/main. Pass the ref as a prop. Currently looks really messy.

Reply:
On second thought, you don't need to pass ref. Just do it in another component and it should work